packages/next-extraction-plugin/src/lib/next-extraction-plugin.ts

```diff
@@ -1,3 +1,107 @@
-export function nextExtractionPlugin(): string {
-  return 'next-extraction-plugin';
+import { GriffelCSSExtractionPlugin } from '@griffel/webpack-extraction-plugin';
+import * as browserslist from 'browserslist';
+import { lazyPostCSS } from 'next/dist/build/webpack/config/blocks/css';
+import { getGlobalCssLoader } from 'next/dist/build/webpack/config/blocks/css/loaders';
+
+import type { GriffelCSSExtractionPluginOptions } from '@griffel/webpack-extraction-plugin';
+import type { NextConfig } from 'next';
+import type { ConfigurationContext } from 'next/dist/build/webpack/config/utils';
+import type { Configuration, RuleSetRule } from 'webpack';
+
+type GriffelNextExtractionPluginOptions = GriffelCSSExtractionPluginOptions & {
+  extractLoaderRuleAttrs?: Omit<RuleSetRule, 'test' | 'use'>;
+};
+
+function getSupportedBrowsers(dir: string, isDevelopment: boolean) {
+  let browsers: string[] | undefined;
+  try {
+    browsers = browserslist.loadConfig({
+      path: dir,
+      env: isDevelopment ? 'development' : 'production',
+    });
+  } catch {
+    // Prevent error on build time
+  }
+
+  return browsers;
 }
+
+export const withGriffelCSSExtraction =
+  ({
+    extractLoaderRuleAttrs: webpackLoaderRules = {},
+    ...webpackPluginOptions
+  }: GriffelNextExtractionPluginOptions = {}) =>
+  (nextConfig: NextConfig = {}) =>
+    Object.assign({}, nextConfig, {
+      webpack(config: Configuration & ConfigurationContext, options) {
+        const { dir, dev, isServer } = options;
+
+        if (!options.defaultLoaders) {
+          throw new Error(
+            'This plugin is not compatible with Next.js versions below 5.0.0 https://err.sh/next-plugins/upgrade',
```

Comment:
A bit confusing as README mentions `NextJS 12.0.5`. Should we remove this statement?

Reply:
I think it's better to leave the test to be safe. Most NextJS plugins have it.

[next-image](https://github.com/twopluszero/next-images/blob/cb101891b5e0eaabf1612d1b6bb518cbd96d72e5/index.js#L19)
[next-fonts](https://github.com/rohanray/next-fonts/blob/9499eedd1f3639fbe1915e1bd7b153abcf764f7f/index.js#L8)

Reply:
Okay, let's keep it 👍